doc/developer/design/20220718_envelopes.md

```diff
@@ -0,0 +1,208 @@
+
+    There is no guarantee of consistency across multiple sources, because
+    there is no means to express such relationships in the data; it would require us creating
+    a language for users to tell us themselves what timestamps their data has.
+  * For `DEBEZIUM` and `DEBEZIUM UPSERT` sources with no transaction metadata
+    topic (or if we never figure out how to read that topic correctly and performantly),
+	the results are _eventually consistent_: if the value for key `k` stops changing
+	in the upstream data, then the value reported by Materialize for key `k` will
+	eventually converge to its true value in the upstream database.
+  * For `DEBEZIUM` and `DEBEZIUM UPSERT` sources _with_ a transaction metadata topic
+    (assuming we are able to implement the corresponding feature), the results are
+	fully consistent: the state observed in Materialize at any timestamp `t` corresponds
+	to the committed result of some transaction in the upstream database.
+
+**Hard Requirement**: Materialize is _robust_: although the result
+of ingesting arbitrarily corrupted data is unspecified, the damage
+should be scoped to the affected set of sources. It should not cause Materialize to crash,
+nor should it cause general undefined behavior (e.g., incorrect results when querying
+unrelated sources or views).
+
+## Non-Goals
+
+It is a permanent non-goal
+to support arbitrarily corrupted data, or to support the classic `DEBEZIUM` envelope in
+situations that do not fit the constraints described above. The famous quote by
+Charles Babbage applies here.
+
+It is a permanent non-goal to support the exotic custom deduplication options
+that formerly existed for `DEBEZIUM` sources, like `full` or `full_in_range`.
+These were designed to account for the specific requirements of one particular
+customer, and were never documented nor widely supported or used; furthermore, nobody
+presently working at Materialize understands their intended semantics in detail.
+Although the one customer they were designed for is indeed important, we
+were happily able to confirm recently that they are able to use `DEBEZIUM UPSERT`
+for all of their topics, making the issue moot.
+
+It is _currently_ a non-goal to support Microsoft SQL Server in the classic `DEBEZIUM` envelope,
+due to [this issue](https://issues.redhat.com/browse/DBZ-3375) preventing us from
+precisely duplicating records. If we come up with a way to deduplicate records that we're
+confident in, we will re-evaluate this. Note that Debezium data from
+Microsoft SQL Server should work
+fine with `DEBEZIUM UPSERT`
+
+Solving the issue of [bounded input reliance](https://github.com/MaterializeInc/materialize/issues/13534)
+is outside the scope of this project; until that issue is solved, we must assume
+that data is not being compacted away while we're busy reading it, which is
+somewhat problematic as we have no way to express our frontiers back upstream to
+the user's Kafka installation.
```

Comment:
Can you say more about what this means? This project does not attempt to avoid any reliance on the input, or that given that it is hard to communicate this to Kafka, .. no promises? Both could be fine, but e.g. bounded input reliance for `UPSERT` seems probably not that hard maybe? Idk.

Reply:
My impression is that someone is thinking about this issue, and that person isn't me. So, "no promises as part of this project" is the correct interpretation.